Common/src/main/java/at/petrak/hexcasting/api/casting/mishaps/MishapBad.kt

```diff
@@ -0,0 +1,24 @@
+package at.petrak.hexcasting.api.casting.mishaps
+
+import com.mojang.datafixers.util.Either
+import net.minecraft.core.BlockPos
+import net.minecraft.world.entity.Entity
+import net.minecraft.world.item.ItemStack
+import net.minecraft.world.phys.Vec3
+
+// a silly stupid wrapper for throwing arbitrary badness mishaps (the ones that have a .of(thing, stub) constructor.
+class MishapBad {
```

Comment:
IMO this should just be a function in ActionUtils or something. Or if it has to be its own class for some reason, it should at least be abstract.